tests/unit/acquisition/test_function.py

```diff
@@ -85,14 +90,37 @@ def test_expected_improvement_builder_builds_expected_improvement(
     npt.assert_array_almost_equal(acq_fn(query_at), expected)
 
 
-def test_expected_improvement() -> None:
-    def _ei(x: tf.Tensor) -> tf.Tensor:
-        n = tfp.distributions.Normal(0, 1)
-        return - x * n.cdf(-x) + n.prob(-x)
+@random_seed()
+@pytest.mark.parametrize('best, rtol, atol', [
+    (tf.constant([50.0]), 0.01, 1e-3),
+    (BRANIN_GLOBAL_MINIMUM[None], 0.01, 1e-3),
+    (BRANIN_GLOBAL_MINIMUM[None] * 1.01, 0.01, 1e-3)
+])
+@pytest.mark.parametrize('variance_scale', tf.constant([0.1, 1.0, 10.0], dtype=tf.float64))
+def test_expected_improvement(
+    variance_scale: tf.Tensor, best: tf.Tensor, rtol: float, atol: float
+) -> None:
+    best = tf.cast(best, dtype=tf.float64)
+
+    x_range = tf.linspace(0.0, 1.0, 11)
+    x_range = tf.cast(x_range, dtype=tf.float64)
+    xs = tf.reshape(tf.stack(tf.meshgrid(x_range, x_range, indexing='ij'), axis=-1), (-1, 2))
+
+    class _Model(GaussianMarginal):
+        kernel = tfp.math.psd_kernels.MaternFiveHalves(variance_scale, length_scale=0.25).apply
+
+        def predict(self, query_points: TensorType) -> Tuple[TensorType, TensorType]:
+            return branin(query_points), self.kernel(query_points, query_points)[:, None]
+
+    num_samples_per_point = 1_000_000
```

Comment:
this seems very very high.1_000 should be sufficient (depends on the tolerance of course)

Reply:
doesn't hurt if the test is fast though

Reply:
1000 wasn't passing. is that a problem?

the test is getting fast enough at 1,000,000 unless you run it in pycharm (which for some reason is much slower)

Reply:
Maybe the problem comes from the relative tolerance, the relative error can be quite large for values very close to zero. I would be in favour of only using the absolute tolerance.

Reply:
it's the high variance tests that need lots of samples

Reply:
haha, the 0.1 variance test works with 4 samples

Reply:
try 1_000 with 100 variance, increase to 10_000 if needed

Reply:
i can get it down to 100_000. The values you ask for generally don't pass. Maybe worth investigating if this is a problem in the test/implementation